src/jdk.jshell/share/classes/jdk/internal/jshell/tool/ConsoleIOContext.java

```diff
@@ -1431,6 +1431,14 @@ public TestTerminal(InputStream input, OutputStream output) throws Exception {
         private TestTerminal(InputStream input, OutputStream output, Size size) throws Exception {
             super(input, output, "ansi", size, size);
         }
+
+        @Override
+        public Attributes enterRawMode() {
+            Attributes res = super.enterRawMode();
+            res.setControlChar(ControlChar.VEOF, 4);
```

Comment:
This is to correctly recognize `4`/Ctrl-D in the input in tests. Note this terminal is supposed to only be used by tests.